src/backend/distributed/commands/statistics.c

```diff
@@ -169,7 +170,18 @@ PreprocessDropStatisticsStmt(Node *node, const char *queryString,
 		return NIL;
 	}
 
-	QualifyTreeNode((Node *) dropStatisticsStmt);
+	if (!CheckStatisticsOwnershipValid(dropStatisticsStmt))
```

Comment:
Again, my question is the following: Why do we have this problem only for drop statistics objects? Say, not for schema objects or any other objects?

I'm mostly asking for this: Could/should we add a more generic check to the `DistOpsValidityState`?

Or, is this really about statistics objects?

Reply:
It can be generalized for dropping objects. 1 test failed because pg tested that flow only for statistics object.